File: rules/java/weak-ciphers.java
// Test fixtures for pqc-semgrep-rules / Java / classically-broken ciphers.
import javax.crypto.Cipher;

public class WeakCiphersFixture {
    public static void main(String[] args) throws Exception {
        // ruleid: pqc.java.des
        Cipher des1 = Cipher.getInstance("DES");
        // ruleid: pqc.java.des
        Cipher des2 = Cipher.getInstance("DES/CBC/PKCS5Padding");

        // ruleid: pqc.java.3des
        Cipher des3 = Cipher.getInstance("DESede");
        // ruleid: pqc.java.3des
        Cipher des4 = Cipher.getInstance("DESede/CBC/PKCS5Padding");

        // ruleid: pqc.java.rc4
        Cipher rc4_1 = Cipher.getInstance("RC4");
        // ruleid: pqc.java.rc4
        Cipher rc4_2 = Cipher.getInstance("ARCFOUR");

        // ok: pqc.java.des
        Cipher safe = Cipher.getInstance("AES/GCM/NoPadding");
    }
}

File: rules/javascript/weak-ciphers.js
// Test fixtures for pqc-semgrep-rules / JavaScript / classically-broken ciphers.
const crypto = require('crypto');

function vulnerable() {
    // ruleid: pqc.javascript.des
    crypto.createCipheriv('des-cbc', Buffer.alloc(8), Buffer.alloc(8));

    // ruleid: pqc.javascript.3des
    crypto.createCipheriv('des-ede3-cbc', Buffer.alloc(24), Buffer.alloc(8));

    // ruleid: pqc.javascript.rc4
    crypto.createCipheriv('rc4', Buffer.alloc(16), null);
}

function safe() {
    // ok: pqc.javascript.des
    crypto.createCipheriv('aes-256-gcm', Buffer.alloc(32), Buffer.alloc(12));
}
